src/Microsoft.AspNet.Mvc.Core/ContentResult.cs

```diff
@@ -47,32 +47,31 @@ public override Task ExecuteResultAsync(ActionContext context)
             var logger = loggerFactory.CreateLogger<ContentResult>();
 
             var response = context.HttpContext.Response;
-            var contentTypeHeader = ContentType;
 
-            if (contentTypeHeader != null && contentTypeHeader.Encoding == null)
-            {
-                // Do not modify the user supplied content type, so copy it instead
-                contentTypeHeader = contentTypeHeader.Copy();
-                contentTypeHeader.Encoding = Encoding.UTF8;
-            }
+            string resolvedContentType = null;
+            Encoding resolvedContentTypeEncoding = null;
+            ResponseContentTypeHelper.ResolveContentTypeAndEncoding(
```

Comment:
You don't need to initialize these variables. `out` variables are assigned when you call the function.